doc/rst/high_bandwidth_endpoints.rst

```diff
@@ -0,0 +1,146 @@
+|newpage|
+
+************************************
+High bandwidth isochronous endpoints
+************************************
+
+``lib_xud`` provides support for high bandwidth isochronous endpoints on high-speed USB devices, as described in the
+Section 5.9.2 of the `USB 2.0 specification <https://usb.org/document-library/usb-20-specification>`_.
+This enables higher data throughput by allowing multiple transactions per microframe.
+
+The current implementation in ``lib_xud`` has the following limitations:
+
+* A maximum of 2 transactions per microframe is supported (the USB specification allows up to 3).
+* The maximum data payload size per transaction is the same across all high bandwidth endpoints
+* The maximum data payload size per transaction must be a multiple of 4 bytes.
+* This feature is supported only on `xcore.ai` series devices running at an increased system frequency of 800 MHz due to the performance demands of high bandwidth transfers.
+
+Enabling high bandwidth ISO endpoint support
+============================================
+
+High bandwidth ISO endpoint support is disabled by default. To enable it, the application must define the
+``XUD_USB_ISO_MAX_TXNS_PER_MICROFRAME`` macro as ``2``. This allows up to two transactions per microframe.
+By default, this macro is set to ``1`` in ``xud.h``, limiting operation to a single transaction and disabling
+high bandwidth transfers. Applications can override the default by re-defining the macro in a custom ``xud_conf.h`` file.
+
+The ``XUD_USB_ISO_EP_MAX_TXN_SIZE`` macro, also defined in ``xud.h``, sets the maximum payload size per ISO transaction.
+Its default value is ``1024`` bytes. This can also be overridden in the application's ``xud_conf.h`` file if needed.
+
+High bandwidth ISO transfer support is communicated to the host via the ``wMaxPacketSize`` field in the
+Standard AS Isochronous Audio Data Endpoint Descriptor. This field encodes both the maximum packet size
+per transaction and the number of transactions per microframe. The format of this field is defined in
+Table 9-13 of the `USB 2.0 specification <https://usb.org/document-library/usb-20-specification>`_.
+Ensure that ``wMaxPacketSize`` is set correctly in the descriptor to reflect the desired max transaction length and
+number of transactions per microframe.
+
+
+High bandwidth transfer API usage
+=================================
+
+The ``lib_xud`` APIs used for handling high bandwidth transfers are:
+
+* Asynchronous operations:
+  ``XUD_SetReady_Out``, ``XUD_SetReady_In``, ``XUD_GetData_Select``, ``XUD_SetData_Select``
+
+* Synchronous operations:
+  ``XUD_GetBuffer``, ``XUD_SetBuffer``
+
+Asynchronous operation
+----------------------
+
+To initiate an IN or OUT transfer, call ``XUD_SetReady_In`` or ``XUD_SetReady_Out``, respectively.
+This process is identical to that of standard (non-HiBW) isochronous transfers, except:
+
+* These functions are called **once per transfer**, not per transaction.
```

Comment:
add (i.e. once per microframe) for clarity maybe?

Reply:
done